forcedotcom/salesforcedx-vscode-automation-tests
# chore: Deploy and Retrieve

### What does this PR do?
Implementation for Deploy and Retrieve

### What issues does this PR fix or reference?
@W-13068481@

See also https://salesforce.quip.com/qVEkABuGd7ao and https://docs.google.com/spreadsheets/d/1DwUs_Xb-bGB-1XKZsRurN2l3O6Uxq5n68QizkWRNcSI/edit#gid=0

### How to test:

1. Go to the [End to End Tests Workflow](https://github.com/forcedotcom/salesforcedx-vscode/actions/workflows/e2e.yml) in salesforcedx-vscode repo
2. Paste the PR's branch (`cristi/deploy-and-retrieve`) in the second field of the Run workflow dropdown (**Set the branch to use for automation tests**)
3. Choose deployAndRetrieve.e2e.ts from the dropdown to select the test to run.

AC:
All steps within the testsuite should pass.

Successful runs: 
- [5616795350](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5616795350) ✅ 
- [5662691022](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5662691022) ✅
